#! /usr/bin/env python

PACKAGE='wallie_one'

from dynamic_reconfigure.parameter_generator_catkin import *
from driver_base.msg import SensorLevels

gen = ParameterGenerator()

gen.add("A", double_t, SensorLevels.RECONFIGURE_CLOSE,
        "A distortion parameter", 0.1, -3, 3)
gen.add("B", double_t, SensorLevels.RECONFIGURE_CLOSE,
        "B distortion parameter", 0.0, -2, 2)
gen.add("C", double_t, SensorLevels.RECONFIGURE_CLOSE,
        "C distortion parameter", 0.1, -2, 2)
gen.add("D", double_t, SensorLevels.RECONFIGURE_CLOSE,
        "D distortion parameter", .94, -2, 2)

gen.add("resolution", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "resolution", 2, 2, 3)

gen.add("pointCloudFPS", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "pointCloudFPS in operator mode", 40, 1, 100)


gen.add("voxel_size", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Voxel Size in meters.", 0.05, 0.0001, 0.075)


gen.add("op_control", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "Operator control enable toggle.", False)

gen.add("thresh", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Radius of search around point.", .1, 0, 20)

gen.add("radius", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Pixel radius for opencv circle points.", 5, 0, 20)

gen.add("thresh_scaler", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Radius of search around point scalar for VR visualization", 1, 0, 100)
			
gen.add("dist_threshold", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Decimation Factor", 10, 0, 20)
	
gen.add("use_frustum", bool_t, SensorLevels.RECONFIGURE_RUNNING,
        "use_frustum culling.", False)

gen.add("far_plane_distance_viz", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "far_plane_distance_viz", 15, 0, 100)
gen.add("near_plane_distance_viz", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "near_plane_distance_viz ", 10, 0, 50)
gen.add("horizontal_fov_viz", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "horizontal_fov_viz", 90, 0, 200)
gen.add("vertical_fov_viz", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "vertical_fov_viz", 60, 0, 200)

gen.add("far_plane_distance", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "far_plane_distance ", 15, 0, 100)
gen.add("near_plane_distance", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "near_plane_distance ", 10, 0, 50)
gen.add("horizontal_fov", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "horizontal_fov ", 90, 0, 200)
gen.add("vertical_fov", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "vertical_fov ", 60, 0, 200)
	
gen.add("max_z", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Max z for VR visualization", 1, 0, 100)		
gen.add("min_cluster_size", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Decimation Factor", 500, 0, 1000)
	
gen.add("region_color_threshold", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Decimation Factor", 1, 0, 100)
	
gen.add("point_color_threshold", int_t, SensorLevels.RECONFIGURE_RUNNING,
        "Decimation Factor", 1, 0, 100)
		
gen.add("target_frame", str_t, SensorLevels.RECONFIGURE_RUNNING,
        "Target frame for data transform. Defaults to hmd_imu_frame.", "hmd_imu_frame")

gen.add("map_res", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Mapping resolution in meters. Zed node param", 0.1, 0.01, 0.2)

gen.add("fps", double_t, SensorLevels.RECONFIGURE_RUNNING,
        "Frames Per Second (FPS)", 60., 0., 120.)

gen.add("load_custom_pcd", str_t, SensorLevels.RECONFIGURE_CLOSE,
        "Load from a custom pcd file.",
        "temp_cloud.pcd")

gen.add("frame_id", str_t, SensorLevels.RECONFIGURE_CLOSE,
        "ROS tf frame of reference, resolved with tf_prefix unless absolute.",
        "camera")

exit(gen.generate(PACKAGE, "wallie_one", "ZedPubStereo"))

